### DISCUSSION TITLE
rust-course/关于”字符所占的字节数是变长的(1-4)“ 如何理解呢？

### DISCUSSION THREAD

**Discussion by AllenYu0118** [sunface/rust-course#321] [内容建议]
第二章：Rust 基础入门 - 2.4 - 2.41 小节的 ”[什么是字符串](https://course.rs/basic/compound-type/string-slice.html#%E4%BB%80%E4%B9%88%E6%98%AF%E5%AD%97%E7%AC%A6%E4%B8%B2)“ 中：

> 在字符串中不一样，字符串是UTF8编码，也就是字符所占的字节数是变长的(1-4)，这样有助于大幅降低字符串所占用的内存空间

上面的 ”变长的(1-4)“ 如何理解呢？是不是指 ”变化的“ ，还是什么呢？

**Comment by sunface** [comment]
比如字符串 "a我" 由两个字符组成，其中 `'a'` 是 ascii 字符，因此占 1 个字节， `'我‘` 是中文字符，占 3 个字节， 这些是由[UTF8编码规则](https://baike.baidu.com/item/UTF-8/481798)规定的。

**Reply by AllenYu0118** [reply]
那我理解是不是把：
> 也就是字符所占的字节数是变长的(1-4)

修改成：

> 也就是字符所占的字节数是变化的(1-4)

也就是把 ”变长的“ 修改为 ”变化的“ 会不会更准确一些？

**Reply by sunface** [reply]
是的。。没错

**Reply by AllenYu0118** [reply]
感谢 ^_^
